src/main/java/seedu/address/logic/commands/GenerateLocationsCommand.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelPredicate;
+import seedu.address.model.ReadOnlyVisitBook;
+import seedu.address.model.VisitBook;
+
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class GenerateLocationsCommand extends Command {
+
+    public static final String COMMAND_WORD = "generateLocations";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all locations which a person of "
+            + "the specified id (case-insensitive) visited and displays them as a list of locations.\n"
+            + "Parameters: PERSONID ...\n"
```

Comment:
The "..." seems to imply that the command can accept multiple personids. Is that intended?